### PR TITLE
Tools: update Python coding style

### PR DESCRIPTION
**Pull request by khancyr**
Following https://github.com/ArduPilot/ardupilot/pull/4423 , this update all python file in Tools/autotest .
Mostly cosmetic changes, 
Code correction will follow

### CODE REVIEW

File: Tools/autotest/apmrover2.py [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
+import os
+import shutil
+
+import pexpect
 from pymavlink import mavutil
-import random
+
+from common import (expect_callback, expect_list_clear, expect_list_extend,
+                    idle_hook, log_download, message_hook, wait_distance,
+                    wait_heading, wait_location, wait_mode, wait_seconds, wait_waypoint)
```
Comment (OXINARF) [inline Tools/autotest/apmrover2.py]: Can't we ignore this rule? I understand the why, but this is ugly.

File: Tools/autotest/apmrover2.py [hunk 2]
```diff
@@ -1,37 +1,45 @@
...
-    mavproxy.expect('ARMED')
+def arm_rover(mavproxy, mav):		
+    # wait for EKF and GPS checks to pass		
+    wait_seconds(mav, 30)		
+
+    mavproxy.send('arm throttle\n')		
+    mavproxy.expect('ARMED')		
 
-    print("ROVER ARMED")
+    print("ROVER ARMED")
```
Comment (OXINARF) [inline Tools/autotest/apmrover2.py]: What is this? Adding tabs at the end of a line? Why?

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -590,9 +614,10 @@ def fly_gps_glitch_auto_test(mavproxy, mav, timeout=120):
     # record time and position
     tstart = get_sim_time(mav)
     tnow = tstart
+    start_pos = sim_location(mav)
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: Where does this come from?

File: Tools/autotest/apmrover2.py (hunk 1 above)
Comment (khancyr) [inline Tools/autotest/apmrover2.py]: yes we can!

File: Tools/autotest/apmrover2.py (hunk 2 above)
Comment (khancyr) [inline Tools/autotest/apmrover2.py]: Because ... I don't know ... but will kill it

File: Tools/autotest/arducopter.py (hunk 3 above)
Comment (khancyr) [inline Tools/autotest/arducopter.py]: false copy ... shouldn't be add...

File: Tools/autotest/arducopter.py [hunk 4]
```diff
@@ -936,29 +972,29 @@ def fly_ArduCopter(binary, viewerip=None, map=False, valgrind=False, gdb=False):
...
-    mavproxy = util.start_MAVProxy_SIL('ArduCopter', options=options)
+    mavproxy = util.start_MAVProxy_SITL('ArduCopter', options=options)
     mavproxy.expect('Telemetry log: (\S+)')
     logfile = mavproxy.match.group(1)
     print("LOGFILE %s" % logfile)
 
     buildlog = util.reltopdir("../buildlogs/ArduCopter-test.tlog")
     print("buildlog=%s" % buildlog)
-    copyTLog = False
+    copy_t_log = False
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: I would say `copy_tlog` would be preferred

File: Tools/autotest/arducopter.py [hunk 5]
```diff
@@ -1241,38 +1276,38 @@ def fly_ArduCopter(binary, viewerip=None, map=False, valgrind=False, gdb=False):
...
         return False
     return True
 
 
-def fly_CopterAVC(binary, viewerip=None, map=False, valgrind=False, gdb=False):
-    '''fly ArduCopter in SIL for AVC2013 mission
-    '''
+def fly_CopterAVC(binary, viewerip=None, use_map=False, valgrind=False, gdb=False):
+    """Fly ArduCopter in sitl for AVC2013 mission.
+    """
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: Change this to the previous line?

File: Tools/autotest/arduplane.py [hunk 6]
```diff
@@ -1,28 +1,38 @@
-# fly ArduPlane in SIL
+# fly ArduPlane in sitl
```
Comment (OXINARF) [inline Tools/autotest/arduplane.py]: sitl -> SITL

File: Tools/autotest/arduplane.py [hunk 7]
```diff
@@ -426,24 +441,24 @@ def fly_mission(mavproxy, mav, filename, height_accuracy=-1, target_altitude=Non
     return True
 
 
-def fly_ArduPlane(binary, viewerip=None, map=False, valgrind=False, gdb=False):
-    '''fly ArduPlane in SIL
+def fly_ArduPlane(binary, viewerip=None, use_map=False, valgrind=False, gdb=False):
+    """fly ArduPlane in sitl
```
Comment (OXINARF) [inline Tools/autotest/arduplane.py]: sitl -> SITL

**Comment by khancyr** [comment]
Another point : I remove sys.path.insert(0, os.path.join(os.path.dirname(os.path.realpath(__file__)), 'pysim')) that are use to find util in pysim in favor of the use of __init__.py and relative import

File: Tools/autotest/autotest.py [hunk 8]
```diff
@@ -484,18 +511,13 @@ def run_tests(steps):
...
     for a in args:
         arg_matched = False
-        for s in steps:
+        for s in steps_list:
             if fnmatch.fnmatch(s.lower(), a.lower()):
                 matched.append(s)
-                arg_matched = True
-        if not arg_matched:
-            print("No steps matched argument ({})".format(a))
-            sys.exit(1)
```
Comment (OXINARF) [inline Tools/autotest/autotest.py]: Why was this removed?

File: Tools/autotest/common.py [hunk 9]
```diff
@@ -1,83 +1,99 @@
...
         util.pexpect_drain(p)
 
+
 def get_distance(loc1, loc2):
-    '''get ground distance between two locations'''
-    dlat 		= loc2.lat - loc1.lat
-    dlong		= loc2.lng - loc1.lng
+    """Get ground distance between two locations."""
+    dlat        = loc2.lat - loc1.lat
+    dlong       = loc2.lng - loc1.lng
```
Comment (OXINARF) [inline Tools/autotest/common.py]: A single space after dlong and align above? I don't see no reason for all this space.

File: Tools/autotest/quadplane.py [hunk 10]
```diff
@@ -40,23 +46,23 @@ def fly_mission(mavproxy, mav, filename, fence, height_accuracy=-1):
     return True
 
 
-def fly_QuadPlane(binary, viewerip=None, map=False, valgrind=False, gdb=False):
-    '''fly QuadPlane in SIL
+def fly_QuadPlane(binary, viewerip=None, use_map=False, valgrind=False, gdb=False):
+    """Fly QuadPlane in sitl.
```
Comment (OXINARF) [inline Tools/autotest/quadplane.py]: sitl -> SITL

File: Tools/autotest/pysim/util.py [hunk 11]
```diff
@@ -101,20 +120,22 @@ def build_examples(board, j=None, debug=False, clean=False):
...
     return True
 
 
 # list of pexpect children to close on exit
 close_list = []
 
+
 def pexpect_autoclose(p):
-    '''mark for autoclosing'''
+    """mark for autoclosing"""
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: Why this one doesn't start with capital letter?

File: Tools/autotest/pysim/util.py [hunk 12]
```diff
@@ -128,44 +149,47 @@ def pexpect_close(p):
...
+
 def cmd_as_shell(cmd):
-    return (" ".join([ '"%s"' % x for x in cmd ]))
+    return (" ".join(['"%s"' % x for x in cmd]))
+
 
-import re
 def make_safe_filename(text):
-    '''return a version of text safe for use as a filename'''
+    """return a version of text safe for use as a filename"""
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: Same here, why not capital letter?

File: Tools/autotest/pysim/util.py [hunk 13]
```diff
@@ -329,73 +360,76 @@ def BodyRatesToEarthRates(dcm, gyro):
...
-    '''return bearing between two points in degrees, in range 0-360
-    thanks to http://www.movable-type.co.uk/scripts/latlong.html'''
+    """Return bearing between two points in degrees, in range 0-360
+    thanks to http://www.movable-type.co.uk/scripts/latlong.html ."""
     lat1 = math.radians(lat1)
     lat2 = math.radians(lat2)
     lon1 = math.radians(lon1)
     lon2 = math.radians(lon2)
-    dLat = lat2 - lat1
+    # dLat = lat2 - lat1
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: I guess you commented this because it isn't used? Either there is something wrong here and needs to be checked or if really isn't needed then we should remove it.

File: Tools/autotest/pysim/util.py [hunk 14]
```diff
@@ -412,26 +446,25 @@ def __init__(self, windstring, cross_section=0.1):
         self.turbulance_mul = 1.0
 
     def current(self, deltat=None):
-        '''return current wind speed and direction as a tuple
-        speed is in m/s, direction in degrees
-        '''
+        """Return current wind speed and direction as a tuple
+        speed is in m/s, direction in degrees .
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: Why the space after degrees?

File: Tools/autotest/param_metadata/htmlemit.py [hunk 15]
```diff
@@ -1,17 +1,20 @@
 #!/usr/bin/env python
+"""
+Emit docs in a form acceptable to the APM wordpress docs site
```
Comment (OXINARF) [inline Tools/autotest/param_metadata/htmlemit.py]: @peterbarker Is this still true?

Even if it is, this should be ArduPilot instead of APM.

File: Tools/autotest/param_metadata/param_parse.py [hunk 16]
```diff
@@ -1,14 +1,18 @@
...
+import glob
+import os
+import re
+import sys
+from optparse import OptionParser
 
-from param import *
-from wikiemit import WikiEmit
-from xmlemit import XmlEmit
 from htmlemit import HtmlEmit
```
Comment (OXINARF) [inline Tools/autotest/param_metadata/param_parse.py]: I get that this is ordered alphabetically but it makes more sense that _htmlemit_ is together with the other *Emit*s

File: Tools/autotest/param_metadata/wikiemit.py [hunk 17]
```diff
@@ -17,56 +19,53 @@ def __init__(self):
...
         self.emit_comment("Libraries")
-        
-    def emit(self, g, f):    
-        
-        t = "\n\n== %s Parameters ==\n"  % (self.camelcase_escape(g.name))
-        
+
+    def emit(self, g, f):
+
+        t = "\n\n== %s Parameters ==\n" % (self.camelcase_escape(g.name))
```
Comment (OXINARF) [inline Tools/autotest/param_metadata/wikiemit.py]: I haven't checked all places but I think most don't leave a space after function definition so let's remove it here

File: Tools/autotest/jsb_sim/runsim.py [hunk 18]
```diff
@@ -1,13 +1,25 @@
 #!/usr/bin/env python
-# run a jsbsim model as a child process
-
-import sys, os, pexpect, socket
-import math, time, select, struct, signal, errno
+"""
+ Eun a jsbsim model as a child process.
```
Comment (OXINARF) [inline Tools/autotest/jsb_sim/runsim.py]: Eun -> Run

File: Tools/autotest/jsb_sim/runsim.py [hunk 19]
```diff
@@ -162,12 +178,10 @@ def process_jsb_input(buf, simtime):
     try:
         sim_out.send(simbuf)
     except socket.error as e:
-        if e.errno not in [ errno.ECONNREFUSED ]:
+        if e.errno not in [errno.ECONNREFUSED]:
             raise
 
 
-
-##################
```
Comment (OXINARF) [inline Tools/autotest/jsb_sim/runsim.py]: I think this can be left, it's there to visualy separate the main program

**Comment by OXINARF** [comment]
> Another point : I remove sys.path.insert(0, os.path.join(os.path.dirname(os.path.realpath(file)), 'pysim')) that are use to find util.py in pysim in favor of the use of init.py and relative import

And I very much prefer this new way.

I think the commit for apm_unit_tests shouldn't be applied and instead that folder removed. It was committed 4 years ago and never changed, I'm pretty sure nobody uses it and it is outdated.

File: Tools/autotest/arducopter.py (hunk 4 above)
Comment (khancyr) [inline Tools/autotest/arducopter.py]: agreed

File: Tools/autotest/autotest.py (hunk 8 above)
Comment (khancyr) [inline Tools/autotest/autotest.py]: my mistake (too much rebase...)

File: Tools/autotest/common.py (hunk 9 above)
Comment (khancyr) [inline Tools/autotest/common.py]: I try to not break current indentation (and it really bother me like that case ! ). I will correct that

Comment (OXINARF) [inline Tools/autotest/common.py]: You can break current indentation if it is wrong, that's not a problem.

**Comment by khancyr** [comment]
there is 2 errors in autotest : mavproxy map module isn't load but it try to set :
'map set showgpspos 1'
'map set showsimpos 1'

It cause by True in : https://github.com/ArduPilot/ardupilot/blob/master/Tools/autotest/arducopter.py#L471 and https://github.com/ArduPilot/ardupilot/blob/master/Tools/autotest/arducopter.py#L572

File: Tools/autotest/param_metadata/htmlemit.py [hunk 20]
```diff
@@ -1,17 +1,20 @@
 #!/usr/bin/env python
+"""
+Emit docs in a form acceptable to the Ardupilot wordpress docs site
```
Comment (OXINARF) [inline Tools/autotest/param_metadata/htmlemit.py]: Change _Ardupilot_ to _old ArduPilot_

In the future we should change this to generate a normal html file.

**Comment by OXINARF** [comment]
Those errors, if corrected, should be done in a separate PR.

File: Tools/autotest/param_metadata/htmlemit.py (hunk 15 above)
Comment (peterbarker) [inline Tools/autotest/param_metadata/htmlemit.py]: On Fri, 29 Jul 2016, Francisco Ferreira wrote:

> In Tools/autotest/param_metadata/htmlemit.py:
> 
> > @@ -1,17 +1,20 @@
> >  #!/usr/bin/env python
> > +"""
> > +Emit docs in a form acceptable to the APM wordpress docs site
> 
> @peterbarker Is this still true?

I do not believe so.  I believe this was replaced with the current 
ardupilot "wiki", which uses the rstemit.py output.

**Comment by khancyr** [comment]
PING

**Comment by khancyr** [comment]
@OXINARF 
I update to lastest master

**Comment by OXINARF** [comment]
@khancyr I'll try to review it again this week, sorry for the delay.

File: Tools/autotest/arducopter.py [hunk 21]
```diff
@@ -444,41 +461,43 @@ def fly_fence_test(mavproxy, mav, timeout=180):
...
-    
-# fly_gps_glitch_loiter_test - fly south east in loiter and test reaction to gps glitch
+
+
 def fly_gps_glitch_loiter_test(mavproxy, mav, use_map=False, timeout=30, max_distance=20):
-    '''hold loiter position'''
-    mavproxy.send('switch 5\n') # loiter mode
+    """fly_gps_glitch_loiter_test.
+
+     Fly south east in loiter and test reaction to gps glitch."""
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: Why has this comment gone inside? All others stayed outside the function.

File: Tools/autotest/arducopter.py [hunk 22]
```diff
@@ -647,17 +667,18 @@ def fly_gps_glitch_auto_test(mavproxy, mav, use_map=False, timeout=120):
 
     return ret
 
-#fly_simple - assumes the simple bearing is initialised to be directly north
+
+#   fly_simple - assumes the simple bearing is initialised to be directly north
 #   flies a box with 100m west, 15 seconds north, 50 seconds east, 15 seconds south
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: Single space after #?

File: Tools/autotest/pysim/util.py [hunk 23]
```diff
@@ -329,73 +360,75 @@ def BodyRatesToEarthRates(dcm, gyro):
...
-    lat2 = asin(sin(lat1)*cos(dr) +
-                cos(lat1)*sin(dr)*cos(brng))
-    lon2 = lon1 + atan2(sin(brng)*sin(dr)*cos(lat1), 
-                        cos(dr)-sin(lat1)*sin(lat2))
+    dr = distance / radius_of_earth
+
+    lat2 = asin(sin(lat1) * cos(dr) +
+                cos(lat1) * sin(dr) * cos(brng))
+    lon2 = lon1 + atan2(sin(brng) * sin(dr) * cos(lat1),
+                        cos(dr) - sin(lat1) * sin(lat2))
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: Can we put these two in the same line? I think it would be easier to read, don't you?

File: Tools/autotest/pysim/util.py [hunk 24]
```diff
@@ -329,73 +360,75 @@ def BodyRatesToEarthRates(dcm, gyro):
...
+radius_of_earth = 6378100.0  # in meters
+
 
 def gps_newpos(lat, lon, bearing, distance):
-    '''extrapolate latitude/longitude given a heading and distance 
-    thanks to http://www.movable-type.co.uk/scripts/latlong.html
-    '''
+    """Extrapolate latitude/longitude given a heading and distance
+    thanks to http://www.movable-type.co.uk/scripts/latlong.html .
+    """
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: Move the comment ending to the line before.

**Comment by OXINARF** [comment]
@khancyr Other than those small things this is good to go. Ping me when it's done and I'll ask for merging.

**Comment by magicrub** [comment]
Per @OXINARF 's comment, I marked as reviewed and assigned to @peterbarker

**Comment by peterbarker** [comment]
OK, I'm looking at this now.

@khancyr Can you outline how you generated these changes, please?  I really don't like churning code for formatting reasons (it makes it harder to find what change broke things, and people with out-of-master code must contend with conflicts caused by non-functional changes), so it would be nice to give people a "pre-PR" process to ensure the Python code is up-to-scratch format-wise before it goes in.  This might mean having some sort of exception-to-the-rule mechanism to keep in things we like just-the-way-they-are.  At the very least, 

Most of these changes are straight-forward, and I've pushed most of them in.  I have some objections:

Changing plurality of "file" to "files"; there's only one file...

`+        wait_altitude(mav, (alt_min - 5), alt_min)`
I'd prefer `wait_altitude(mav, (alt_min-5), alt_min)`
... but I left that delta in

```
-        if not wait_heading(mav, 270 - (90*i), accuracy=10):
+        if not wait_heading(mav, 270 - (90 * i), accuracy=10):
```
That's a bad delta, it actively makes that code more difficult to understand.  As above, I'd ordinarily remove the spaces around the "-" sign, too!

Changing "steps" to "steps_list" seems an odd change to make.  Encoding types in names isn't typical in Python?

@khancyr You should be able to rebase your branch on master now and see what I didn't push in, and consider what I really should reconsider :-)

**Comment by khancyr** [comment]
@peterbarker I use flake8 as specified here (http://discuss.ardupilot.org/t/python-code-standard/9128/2) and pylint. But, I don't use auto formating, I make all change by hand otherwise it was worst.

about space aroud operator, I don't like them some much too but all checker tool are reporting them (even in c++ ...) so I surrender...

about file and step_list, I remember correctly, those change where done (not creatively) to remove multiple definition on those name 

I will look a the last change request by you and @OXINARF. 
Now that it is pass, I can submit the same change for mavproxy and pymavlink and then python3 support ! That was the goal !

**Comment by OXINARF** [comment]
@peterbarker I really disagree with that. Math expressions without spaces is one of the worse ways to format code. Those two examples are simple enough to read without spaces, but more complicated math expressions are absolutely impossible to read quickly when there is no spaces between operators.

**Comment by peterbarker** [comment]
On Thu, 1 Sep 2016, Francisco Ferreira wrote:

> @peterbarker I really disagree with that. Math expressions without spaces is
> one of the worse ways to format code. Those two examples are simple enough
> to read without spaces, but more complicated math expressions are absolutely
> impossible to read quickly when there is no spaces between operators.

When the expressions are parameters to functions, I disagree with spaces. 
If it is complicated enough that it is unclear in that case then a temp 
should be created with spaces in it.

Just my opinion, of course.
